performance-es.md

```diff
@@ -183,3 +185,107 @@ de texto, pero el coste es que no será legible para un humano.
 Para la transferencia de datos, cambia a un protocol menos verboso, o
 mejora el API para aceptar consultas parciales. Por ejemplo, usa una consulta
 incremental en lugar de forzar a traer siempre el set de datos completo.
+
+### Cambiar de algoritmo
+
+Si no estás cambiando los datos, la alternativa más importante es cambiar el código.
+
+Es muy posible que las mejoras más importantes vengan de un cambio de algoritmo. Esto es equivalente a sustituir bubble sort (`O(n^2)`) con quicksort (`O(n log n)`) o reemplazar un acceso linear a un array (`O(n)`) con una busqueda binaria (`O(log n)`) o una busqueda en un mapa (`O(1)`).
+
+Así es como el software se vuelve lento. Estructuras originalmente diseñadas para un proposito se reusan para algo que no habian sido diseñadas. Esto ocurre gradualmente.
+
+Es importante tener un entendimiento intuitivo de los diferentes niveles de big-O. Elige la estructura de datos para tu problema. Esto no siempre ahorra ciclos de CPU, pero previene problemas de rendimiento que pueden no ser detectados hasta muchos más adelante.
+
+Las clases básicas de complejidad son:
+
+* O(1): acceso a un campo, un array o un mapa
+
+  Consejo: no te preocupes por ellos
+
+* O(log n): busqueda binaria
+
+  Consejo: sólo es un problema si se hace en un bucle
+
+* O(n): bucle simple
+
+  Consejo: lo haces todo el tiempo
+
+* O(n log n): divide y vencerás, ordenación
+
+  Consejo: sigue siendo bastante rapido
+
+* O(n\*m): bucles anidados / cuadrático
+
+  Consejo: ten cuidado y limita el tamaño de tu conjunto de datos
+
+* Cualquier cosa entre cuadrático y subexponencial
+
+  Consejo: no lo ejecutes en un millón de filas
+
+* O(b ^ n), O(n!): exponencial y mayor
+
+  Consejo: vas a necesitar suerte si tienes más de una o dos docenas de datos
+
+Link: <http://bigocheatsheet.com>
+
+Supongamos que tienes que buscar en un conjunto desordenado de datos. "Debería usar busqueda binaria" piensas, sabiendo que una busqueda binaria es O(log n) que es más rapido que el O(n) de una busqueda linear. Sin embargo, una busqueda binaria requiere que los datos estén ordenados, lo que significa que tendrás que ordenarlos antes, que tarda O(n log n). Si haces muchas busquedas, el coste inicial de la ordenación merecerá la pena. Pero, si sobre todo estas haciendo **lookups**, quizás usar un array fue una decisión equivocada y sería mejor usar un mapa con coste O(1).
+
+Si tu estructura de datos es estática, entonces generalmente podrás hacerlo mucho mejor que en el caso de que fuera dinámica. Resultará más facil construir una estructura de datos óptima para tus patrones de busqueda. Soluciones como minimal perfect hashing pueden tener más sentido aquí, o filtros de Bloom precalculados. Esto también tiene sentido si tu estructura de datos es "estática" durante un periodo largo de manera que puedas amortizar el coste inicial de su construcción en muchas busquedas.
```

Comment:
The English text reads: `If your data structure is static, then you can generally do much better than the dynamic case.`

Based on the context of the text, I believe that `generally do much better` refers to performance or the fact the improvements are a more likely to be done. For that reason, I would suggest the following change:

`s/entonces generalmente podrás hacerlo mucho mejor que/entonces podrás implementar cambios mucho mas eficaces para mejorar el rendimiento que`

Reply:
The intent was that data structures for the static case (build once, never modify) are faster/lower memory than the data structures that need to support updates (the dynamic case).

Reply:
@dgryski thank you for the clarification. Yes, that's what I understood from the text. Perhaps my description was not clear enough. I'll continue thinking of other possible translations and wait for the rest to see what they think.

Reply:
I am not sure about this one.

The way I understand the sentence is that is easier to be faster/lower memory with static data structures that with dynamic ones, but it does not speak about changes. Also, the rest of the paragraph is mostly about how to set your static data structure up.